src/Chainweb/Chainweb.hs

```diff
@@ -495,7 +496,7 @@ validatingMempoolConfig cid v gl mv = Mempool.InMemConfig
     preInsertBatch txs = do
         let hashes = V.map (toPactHash . sfst) txs
         pex <- readMVar mv
-        rs <- _pactLookup pex (Left cid) hashes >>= either throwM pure
+        rs <- _pactLookup pex (Genesis cid) hashes >>= either throwM pure
```

Comment:
I'm not sure if "Genesis" is the correct name for this case when we're looking up by chain id. Paint my bikeshed please